internal/controller/nats/controller.go

```diff
@@ -51,6 +51,7 @@ type Reconciler struct {
 	client.Client
 	controller                  controller.Controller
 	kubeClient                  k8s.Client
+	natsClients                 map[string]Client
```

Comment:
Why do we need to keep all NATS client in-memory?

Reply:
Not to create connecction again and again if for multiple retries in case stream exists, for example. With this we create a connection in the client once and re-use it if exists .
Here is the connection check https://github.com/kyma-project/nats-manager/blob/d944e326c4ea81686a670139f7b456c0b6433a38/internal/controller/nats/client.go#L36